packages/compass-web/src/index.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  CreateNamespacePlugin,
+  DropNamespacePlugin,
+  RenameCollectionPlugin,
+} from '@mongodb-js/compass-databases-collections';
+
+// TODO(COMPASS-7403): only required while these plugins are not converted to the new
+// plugin interface
+activateExplainPlanPluginRoles(globalAppRegistry);
+activateExportToLanguagePluginRoles(globalAppRegistry);
+activateCompassIndexesPluginRoles(globalAppRegistry);
+
+type CompassWebProps = {
+  darkMode?: boolean;
+  connectionString: string;
+} & Pick<
+  React.ComponentProps<typeof WorkspacesPlugin>,
+  'initialWorkspaceTabs' | 'onActiveWorkspaceTabChange'
+>;
+
+const loadingContainerStyles = css({
+  width: '100%',
+  height: '100%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+});
+
+const spinnerStyles = css({
+  flex: 'none',
+});
+
+function LoadingScreen({ connectionString }: { connectionString: string }) {
+  const host = useMemo(() => {
+    try {
+      const url = new ConnectionString(connectionString);
+      return url.hosts[0];
+    } catch {
+      return 'cluster';
+    }
+  }, [connectionString]);
+
+  return (
+    <div data-testid="compass-web-loading" className={loadingContainerStyles}>
+      <SpinLoaderWithLabel
+        className={spinnerStyles}
+        progressText={`Connecting to ${host}…`}
+      ></SpinLoaderWithLabel>
+    </div>
+  );
+}
+
+const DEFAULT_TAB = { type: 'Databases' } as const;
+
+const connectedContainerStyles = css({
+  width: '100%',
+  height: '100%',
+  display: 'flex',
+});
+
+const CompassWeb = ({
+  darkMode,
+  connectionString,
+  initialWorkspaceTabs,
+  onActiveWorkspaceTabChange,
+  // @ts-expect-error not an interface we want to expose in any way, only for
+  // testing purposes, should never be used otherwise
+  __TEST_MONGODB_DATA_SERVICE_CONNECT_FN,
+}: CompassWebProps) => {
+  const [connected, setConnected] = useState(false);
+  const [connectionError, setConnectionError] = useState<any | null>(null);
+  const dataService = useRef<DataService>();
+
+  useEffect(() => {
+    const controller = new AbortController();
+    let ds: DataService;
+    void (async () => {
+      try {
+        const connectFn =
+          (__TEST_MONGODB_DATA_SERVICE_CONNECT_FN as typeof connect) ?? connect;
+        ds = await connectFn({
+          connectionOptions: { connectionString },
+          signal: controller.signal,
+        });
+        dataService.current = ds;
+        setConnected(true);
+      } catch (err) {
+        setConnectionError(err);
+      }
+    })();
+    return () => {
+      void ds?.disconnect();
+    };
+  }, [connectionString, __TEST_MONGODB_DATA_SERVICE_CONNECT_FN]);
+
+  // Re-throw connection error so that parent component can render an
+  // appropriate error screen with an error boundary (only relevant while we are
+  // handling a single connection)
+  if (connectionError) {
+    throw connectionError;
+  }
+
+  if (connected && dataService.current) {
```

Comment:
This is the tiniest nit, but I would probably have treated the inverse (ie. the loading state) as the early return, then all this can be one indentation level less :)

Reply:
Yeah, this makes sense, thanks for the suggestion! To wrap up today and avoid re-running CI again, I'll not change this in this PR, but will do this in a follow up